rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_publisher_info.hpp

```diff
@@ -25,29 +26,36 @@
 #include "rmw/rmw.h"
 
 #include "rmw_fastrtps_shared_cpp/TypeSupport.hpp"
+#include "rmw_fastrtps_shared_cpp/custom_event_info.hpp"
+
 
 class PubListener;
 
-typedef struct CustomPublisherInfo
+typedef struct CustomPublisherInfo : public CustomEventInfo
 {
+  virtual ~CustomPublisherInfo() = default;
+
   eprosima::fastrtps::Publisher * publisher_;
   PubListener * listener_;
   rmw_fastrtps_shared_cpp::TypeSupport * type_support_;
   rmw_gid_t publisher_gid;
   const char * typesupport_identifier_;
+
+  EventListenerInterface * getListener();
```

Comment:
The reason we went with pointers is to be a bit more consistent with how `__rmw_wait()` in `rmw_wait.cpp` has been calling `attachCondition()` for the rest of the `Custom*Info` types.

And it can't be `const` because `attachCondition()` is a non-const member function.